File: redsheet-portal/src/app/database-models/state.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { Country,  } from './'
export class State {
    constructor(state?: State )  {
        let properties = [ "stateId", "stateName", "stateShortName", "statusId", "countryId",];
        for (let property of properties)
            if (state && state[property])
                this[property] = state[property];
    }
 
	stateId : number =   0 ;
 
    @required()
    @maxLength(100)
	stateName : string =   undefined;
 
    @maxLength(10)
	stateShortName : string =   undefined;
 
	statusId : number =   undefined;
 
    @range(0,2147483647)
	countryId : number =   undefined;
	country : Country  ;


}

File: RedSheet.DbEntities/Models/State.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("States", Schema = "dbo")]
	[Serializable]
	public partial class State  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int StateId { get; set; } 
 
	[Required] 
	[MaxLength(100)]
	public string StateName { get; set; } 
 
	[MaxLength(10)]
	public string StateShortName { get; set; } 

	public Nullable<int> StatusId { get; set; } 
 
	[RelationshipTableAttribue("Countries","dbo")]
	//Relationships
	public int CountryId { get; set; } 
 
	[ForeignKey("CountryId")]
	public Country Country { get; set; } 

	public State()
	{
    }

    }
}

